services/QuillLMS/client/app/bundles/Grammar/components/concepts/concept.tsx

```diff
@@ -79,79 +50,20 @@ class Concept extends React.Component<ConceptProps, ConceptState> {
     )
   }
 
-  submit(): void {
-    if (this.state.prompt !== '') {
-      this.props.dispatch(questionActions.submitNewQuestion({
-        prompt: this.state.prompt,
-        concept_uid: this.state.concept_uid,
-        instructions: this.state.instructions,
-        flag: this.state.flag,
-        rule_description: this.state.rule_description,
-        answers: this.state.answers
-      }))
-    }
-  }
-
-  handlePromptChange(e: string) {
-    this.setState({prompt: e})
-  }
-
-  handleInstructionsChange(e: React.ChangeEvent<HTMLTextAreaElement>) {
-    this.setState({instructions: e.target.value})
-  }
-
-  handleRuleDescriptionChange(e: string) {
-    this.setState({rule_description: e})
-  }
-
-  handleSelectorChange(e: {value: string}) {
-    this.setState({concept_uid: e.value})
-  }
-
-  handleConceptChange() {
-    this.setState({concept_uid: this.refs.concept.value})
-  }
-
-  handleFlagChange(e: React.ChangeEvent<HTMLInputElement>) {
-    this.setState({ flag: e.target.value, });
-  }
-
-  handleAnswersChange(e: React.ChangeEvent<HTMLTextAreaElement>) {
-    this.setState({ answers: [{ text: e.target.value }], });
+  submit = (question): void => {
+    this.props.dispatch(questionActions.submitNewQuestion(question))
```

Comment:
We can use `async/await` instead of `dispatch` in many of the existing `dispatch` call patterns, since we're not actually leveraging redux. Beyond the scope of your PR, but worth mentioning